lib/dirac_twisted_clover.cpp

```diff
@@ -175,35 +179,41 @@ namespace quda {
   }
 
   // Public method to apply the inverse twist
-  void DiracTwistedCloverPC::TwistCloverInv(ColorSpinorField &out, const ColorSpinorField &in, const int parity) const
+  void DiracTwistedCloverPC::TwistCloverInv(cvector_ref<ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in,
+                                            QudaParity parity) const
   {
     twistedCloverApply(out, in, QUDA_TWIST_GAMMA5_INVERSE, parity);
   }
 
-  void DiracTwistedCloverPC::WilsonDslash(ColorSpinorField &out, const ColorSpinorField &in, QudaParity parity) const
+  void DiracTwistedCloverPC::WilsonDslash(cvector_ref<ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in,
+                                          QudaParity parity) const
   {
     if (in.TwistFlavor() == QUDA_TWIST_SINGLET) {
       DiracWilson::DslashXpay(out, in, parity, in, 0.0);
     } else {
       // we need an Ls=plain 2 Wilson dslash, which is exactly what the 4-d preconditioned DWF operator is
-      ApplyDomainWall4D(out, in, *gauge, 0.0, 0.0, nullptr, nullptr, in, parity, dagger, commDim.data, profile);
+      for (auto i = 0u; i < in.size(); i++)
```

Comment:
`ApplyDomainWall4D` has been multi-rhs-ified, so we can get rid of the `for` loop here and in the other function below.

Reply:
142ffbe7711caefa9f978bc2405adf3353adb265